### PR TITLE
Generate ProductVersion constant at build time

### LINKED ISSUE #11490: ProductVersion should be generated at compile time
When we rev our product version, we currently have to do it in two places:

https://github.com/NuGet/NuGet.Client/blob/ba01cc633f88fb2dd1bfcbcdd859fb89525a56eb/build/config.props#L13-L17

https://github.com/NuGet/NuGet.Client/blob/ba01cc633f88fb2dd1bfcbcdd859fb89525a56eb/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L85

We could use the values from `config.props` to generate the `ProductVersion` constant in `NuGetPackage.cs`

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11490

Regression? Last working version:

## Description
Use the values in `config.props` to generate a source file to compile the constant on-the-fly

## PR Checklist

- [x] PR has a meaningful title
- [ ] PR has a linked issue.
- [ ] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj [hunk 1]
```diff
@@ -157,4 +157,44 @@
...
+// </auto-generated>
+//------------------------------------------------------------------------------
+namespace $(RootNamespace)
+{
+    public sealed partial class NuGetPackage
+    {
+        /// <summary>
+        /// The current version of the NuGet Visual Studio package which is displayed in the Help - About box of Visual Studio
+        /// </summary>
+        public const string ProductVersion = "$(MajorNuGetVersion).$(MinorNuGetVersion).$(PatchNuGetVersion)"%3B
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj]: You can probably just say `SemanticVersion`.

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj (hunk 1 above)
Comment (jeffkl) [reply src/NuGet.Clients/NuGet.Tools/NuGet.Tools.csproj]: Good idea, I also just updated the comment in `config.props`.

**Review by donnie-msft — APPROVED** [review]

**Review by zivkan — APPROVED** [review]
❤️

**Comment by jeffkl** [comment]
Don't merge until after https://github.com/NuGet/NuGet.Client/pull/4392 is in and this is rebased

**Comment by nkolev92** [comment]
https://github.com/NuGet/NuGet.Client/pull/4392 was merged, you can rebase @jeffkl

**Pull request merged**